configure.ac

```diff
@@ -671,6 +671,10 @@ define(MINIMUM_REQUIRED_BOOST, 1.47.0)
 dnl Check for boost libs
 AX_BOOST_BASE([MINIMUM_REQUIRED_BOOST])
 AX_BOOST_SYSTEM
+if test x$BOOST_SYSTEM_LIB = x; then
+    AC_MSG_ERROR([Boost system library required.])
+fi
+
```

Comment:
I reproduce the problem on current dev (@24c460), using `CXXFLAGS=-std=c++14` generate this error during configure step: 

```
checking for dynamic linked boost test... yes
checking for mismatched boost c++11 scoped enums... ok
configure: error: No working boost sleep implementation found.
```

applying the patch to `configure.ac` locally actually use ther correct error message to notice that libboost system is not found, i.e.